relevanceai/http_client.py

```diff
@@ -100,3 +102,81 @@ def check_auth(self):
             return response.status_code == 200    
         except:
             raise Exception("Invalid auth details.")
+
+    def kmeans_cluster(self,
+        dataset_id: str,
+        vector_fields: list,
+        filters: List = [],
+        k: Union[None, int] = 10,
+        init: str = "k-means++",
+        n_init: int = 10,
+        max_iter: int = 300,
+        tol: float = 1e-4,
+        verbose: bool = True,
+        random_state = None,
+        copy_x: bool = True,
+        algorithm: str ="auto",
+        alias: str = "default",
+        cluster_field: str="_cluster_"
+    ):
+        """
+        This function performs all the steps required for Kmeans clustering:
+        1- Loads the data
+        2- Clusters the data
+        3- Updates the data with clustering info
+        4- Adds the centroid to the hidden centroid collection
+
+        input parameters:
+        dataset_id: name of the dataser
+        vector_fields: a list containing the vector field to be used for clustering
+        filters: a list to filter documents of the dataset,
+        k: K in Kmeans,
+        init: "k-means++" -> Kmeans algorithm parameter ,
+        n_init: number of reinitialization for the kmeans algorithm,
+        max_iter: max iteration in the kmeans algorithm,
+        tol: tol in the kmeans algorithm,,
+        verbose: bool = True,
+        random_state = None,
+        copy_x: bool = True,
+        algorithm: str ="auto",
+        alias: str = "default", string to be used in naming of the field showing the clustering results
+        cluster_field: str="_cluster_"
+
+        returns:
+        a list containing the calculated centroids
+
+        """
```

Comment:
Should users know if they are about to overwrite a cluster? What should be the expected behavior here? 

While the design is up to you, I suggest having an `overwrite=False` be the default and then if the expected output cluster field is already there, it errors and to overwrite it, have `overwrite=True`